### PR TITLE
testing Servlet connection

### PR DESCRIPTION
**Pull request by keyuyan1145**
I modified the file to say "Hello Yuyan!" instead of the default "Hello World!" in order to test Servlet connection.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -27,6 +27,6 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("<h1>Hello Yuyan!</h1>");
```
Comment (rjkroege) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: It doesn't look like you wired up the JavaScript code to actually call the servlet?

File: portfolio/src/main/webapp/experience.html [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		<p><i>March 2019 - May 2020</i></p>
+		<p>
+		As an advising fellow, I advised 2 low-income, high achieving high school 
+		students about college application process for three semesters, 
+		starting with drafting a balanced college list to the final stage of 
+		commiting to a college.
+		</p>
+	</div>
+</body>
+</html>
```
Comment (rjkroege) [inline portfolio/src/main/webapp/experience.html:100]: Could you separate the portfolio content commits into a separate PR from the servlet changes?

File: portfolio/src/main/webapp/project.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	<link rel="stylesheet" href="style.css">
+	<script src="script.js"></script>
+</head>
+
+<body>
+	<div class="nav" id="mynav">
+		<a href="index.html">Home</a>
+		<a class="active" href="project.html">Projects</a>
+		<a href="experience.html">Experiences</a>
+		<a href="javascript:vohid(0);" class="icon" onclick="myFunction()">
```
Comment (rjkroege) [inline portfolio/src/main/webapp/project.html:16]: Where is `vohid` defined?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (keyuyan1145) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: I have not done the js yet. This is only for step 1 of the server tutorial, simply testing that servlet connection is good. But I'm working on the js stuff during later steps.

File: portfolio/src/main/webapp/experience.html (hunk 2 above)
Comment (keyuyan1145) [reply portfolio/src/main/webapp/experience.html:100]: Yes. I committed the portfolio stuff into a different branch. I thought those wouldn't show up under this branch, but they did. Is there any way I could remove those commits from this branch and have the portfolio commits be only visible under the branch named "pages?"

Comment (rjkroege) [reply portfolio/src/main/webapp/experience.html:100]: Did you rebase your new branch from master? After committing here (in github) you need to do something like this:

* `git checkout master` in cloudshell
* `git pull` in cloudshell (to update the `master` branch locally to be up to date with respect to `origin/master` (the branch on GitHub)
* `git checkout <your current banch here>`
* `git rebase master` to make the changes here be w.r.t. master branch

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (rjkroege) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:30]: Ah. OK.

File: portfolio/src/main/webapp/experience.html (hunk 2 above)
Comment (keyuyan1145) [reply portfolio/src/main/webapp/experience.html:100]: I did not rebase from master. So what I did was that I created a new branch for servlet and then went back and committed the portfolio changes under the "pages" branch. Then, I switched back to servlet branch and committed the hello message change. I have not push from the "pages" branch yet. 

Will rebase remove my portfolio commits from the servlet and pages branch, or just the servlet branch?

File: portfolio/src/main/webapp/project.html (hunk 3 above)
Comment (keyuyan1145) [reply portfolio/src/main/webapp/project.html:16]: Sorry, this is a typo, suppose to be void(0) for not doing anything, but I'm changing the syntax to href=#.

File: portfolio/src/main/webapp/experience.html (hunk 2 above)
Comment (rjkroege) [reply portfolio/src/main/webapp/experience.html:100]: You didn't make the pages branch into a PR. Why? Maybe you should fix that.

I think that you created a new `servlet` branch from the `pages` branch. That explains why changes from `pages` are included in the `servlet` branch.

You're welcome to have multiple in-progress PRs but each one should be branched from `master`.

Also don't forget that the branches on GitHub and in your client are not automatically kept in sync. You need to manually synchronize them with `git pull` or `git push`. I suggest that you draw a graph with nodes corresponding to the different branches that you have and think about how actions move state between nodes.

File: portfolio/src/main/webapp/project.html (hunk 3 above)
Comment (rjkroege) [reply portfolio/src/main/webapp/project.html:16]: It's a very good habit (not necessarily that any of us always succeed) to review a change (`git diff`) before committing and test everything that's altered. 🙂

File: portfolio/src/main/webapp/experience.html (hunk 2 above)
Comment (keyuyan1145) [reply portfolio/src/main/webapp/experience.html:100]: I was still working on the portfolio, so I didn't create a PR for it yet. But I just created a separate PR for the portfolio content as well. 

I think that's exactly what happened. I created a branch from pages, instead of master. In this case, could I just delete the servlet branch and recreate it from master?

Thanks for the reminder.

File: portfolio/src/main/webapp/project.html (hunk 3 above)
Comment (keyuyan1145) [reply portfolio/src/main/webapp/project.html:16]: I will keep that in mind. Thanks.

Comment (keyuyan1145) [reply portfolio/src/main/webapp/project.html:16]: The corrected version is pushed under the correct branch now.

File: portfolio/src/main/webapp/experience.html (hunk 2 above)
Comment (rjkroege) [reply portfolio/src/main/webapp/experience.html:100]: Yes. Make a new branch from master. And then you can use `git cherry-pick` to add the last servlet change from this branch into the new branch, push the new branch, etc. and abandon this one.

Comment (keyuyan1145) [reply portfolio/src/main/webapp/experience.html:100]: Thanks for the help. I have deleted this branch and created a new one with the server content.
